icu4c/source/common/udata.cpp

```diff
@@ -206,7 +206,7 @@ setCommonICUData(UDataMemory *pData,     /*  The new common data.  Belongs to ca
     return didUpdate;
 }
 
-#if U_PLATFORM_HAS_WINUWP_API == 0 
+#if U_PLATFORM_HAS_WINUWP_API == 0 || defined(RTC_WINDOWS_UNIVERSAL)
```

Comment:
does it make more sense to define `U_PLATFORM_HAS_WINUWP_API` in the LUA file for builds of WinUWP?  Or more likely have some header file determine if `RTC_WINDOWS_UNIVERSAL` is defined, and if so define `U_PLATFORM_HAS_WINUWP_API`?  

Asking because I don't know how the `U_PLATFORM_HAS_WINUWP_API` is defined.

Reply:
It already is defined. It is needed in other sections of ICU that handle things correctly such as system library loading. Since we do data kinda differently, I opted to not follow these codepaths with runtime specific defines so it can easily be found for future updates. Ideally, a better solution could be found by the ICU maintainers and this could eventually be removed.

Reply:
NM I see we define it in the LUA file, so my incorrect assumptions, sorry.  Basically what you're saying is this protection scheme isn't necessary for WinUWP, and it'd be better if we removed it.  Likely removing it would be more upstream friendly.

Reply:
No, not really. It is supposed to be there for ICU's data workflows. We have an uncommon workflow of statically linking in the data which I don't think they accounted for. We're working around it but a true fix would be much more involved with a potentially new enum value for reading from memory. I did it this way to make us more downstream friendly by changing lines that will most likely not change again so we can easily merge in future updates. I need to open an issue with ICU to let them know to see if there's a way to force memory data so the codepaths that try loading data from disk are not even attempted.